mne/beamformer/_lcmv.py

```diff
@@ -66,65 +66,60 @@ def _setup_picks(picks, info, forward, noise_cov=None):
 
 
 @verbose
-def _apply_lcmv(data, info, tmin, forward, reg, noise_cov=None, 
-                data_cov=None, label=None, picks=None, pick_ori=None, 
-                rank=None, apply_noise_norm=False, weight_norm=None, 
+def _apply_lcmv(data, info, tmin, forward, reg, noise_cov=None,
+                data_cov=None, label=None, picks=None, pick_ori=None,
+                rank=None, apply_noise_norm=False, weight_norm=None,
                 verbose=None):
     """LCMV beamformer for evoked data, single epochs, and raw data."""
     is_free_ori, ch_names, proj, vertno, G = \
         _prepare_beamformer_input(info, forward, label, picks, pick_ori)
 
     # data covariance
     if data_cov is None:
-         raise ValueError('Source reconstruction with beamformers requires '
-                         'a data covariance matrix.')       
+        raise ValueError('Source reconstruction with beamformers requires '
+                         'a data covariance matrix.')
     data_cov = pick_channels_cov(data_cov, include=ch_names)
     Cm = data_cov['data']
 
     # apply SSPs
     if info['projs']:
-      Cm = np.dot(proj, np.dot(Cm, proj.T))
-
+        Cm = np.dot(proj, np.dot(Cm, proj.T))
 
     # check number of sensor types
-    ch_types = [d['coil_type'] for d in info['chs']]
-    ch_types = set(ch_types)
-    
-    if len(ch_types) > 1 and noise_cov is None:
-         raise ValueError('Source reconstruction with several sensor types '
+    ch_types = [_contains_ch_type(info, tt) for tt in ('mag', 'grad', 'eeg')]
+    if sum(ch_types) > 1 and noise_cov is None:
+        raise ValueError('Source reconstruction with several sensor types '
```

Comment:
Mh, doing that here actually doesn't work if picks is set in lcmv(). I guess it needs to be done on the covariance matrix then?